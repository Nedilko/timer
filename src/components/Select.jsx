function Select({Title, Options}) {
  return (
    <div className='py-2'>Day
      <select className="rounded mx-1" name="" id="">
        <option value="1">Monday</option>
        <option value="2">Tuesday</option>
        <option value="3">Wednesday</option>
      </select>
    </div>
  )
}