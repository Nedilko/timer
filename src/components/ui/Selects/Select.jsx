import { useState } from "react";
import PropTypes from "prop-types";

Select.propTypes = {
  label: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.number,
      name: PropTypes.string,
    })
  ),
  selected: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};

function Select({ label, selected, onChange, options }) {
  const [value, setValue] = useState(selected);

  if (!options) return false;

  const handleChange = (e) => {
    onChange(Number(e.target.value));
    setValue(Number(e.target.value));
  };

  return (
    <label role="label" className="py-2 select-none">
      {label}
      <select
        role="listbox"
        onChange={handleChange}
        className="rounded mx-1"
        value={value}
      >
        {options.map((option) => (
          <option role="option" key={option.key} value={option.key}>
            {option.name}
          </option>
        ))}
      </select>
    </label>
  );
}

export default Select;
