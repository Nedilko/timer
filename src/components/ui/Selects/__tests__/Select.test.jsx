import Select from "../Select";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Select", () => {
  const onChange = jest.fn();

  const options = [
    {
      key: 1,
      name: "A",
    },
    {
      key: 2,
      name: "B",
    },
    {
      key: 3,
      name: "C",
    },
  ];

  beforeEach(() => {
    onChange.mockClear();
  });

  it("should match snapshot", () => {
    render(
      <Select
        label="test label"
        selected={1}
        onChange={onChange}
        options={options}
      />
    );

    expect(screen.getByRole("listbox")).toMatchSnapshot();
  });

  it("should have label", () => {
    render(
      <Select
        label="test label"
        selected={1}
        onChange={onChange}
        options={options}
      />
    );

    expect(screen.getByRole("label")).toHaveTextContent("test label");
  });

  it("should have three options", () => {
    render(<Select selected={1} onChange={onChange} options={options} />);

    expect(screen.getAllByRole("option")).toHaveLength(3);
  });

  it("should select option", () => {
    render(<Select selected={1} onChange={onChange} options={options} />);

    userEvent.selectOptions(screen.getByRole("listbox"), ["3"]);

    expect(screen.getByRole("option", { name: "A" }).selected).toBeFalsy();
    expect(screen.getByRole("option", { name: "B" }).selected).toBeFalsy();
    expect(screen.getByRole("option", { name: "C" }).selected).toBeTruthy();

    userEvent.selectOptions(screen.getByRole("listbox"), ["2"]);

    expect(screen.getByRole("option", { name: "A" }).selected).toBeFalsy();
    expect(screen.getByRole("option", { name: "B" }).selected).toBeTruthy();
    expect(screen.getByRole("option", { name: "C" }).selected).toBeFalsy();
  });
});
