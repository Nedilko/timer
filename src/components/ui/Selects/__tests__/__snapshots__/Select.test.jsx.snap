// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot 1`] = `
<select
  class="rounded mx-1"
  role="listbox"
>
  <option
    role="option"
    value="1"
  >
    A
  </option>
  <option
    role="option"
    value="2"
  >
    B
  </option>
  <option
    role="option"
    value="3"
  >
    C
  </option>
</select>
`;
